### PR TITLE
Add VPC network integration tests + fix tags bug in networks

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Feature: testing
- Bugfix

### Detail
Implement tests for Networksapi calls (inside core/vpc + listEnvironmentNetworks) as part of https://github.com/data-dot-all/dataall/issues/1220

++ It also fixes a small bug on networks - tags were not correctly saved in the database!

### Relates
- #1220 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests_new/integration_tests/core/vpc/conftest.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import pytest
+from integration_tests.core.vpc.queries import create_network, delete_network
+
+
+@pytest.fixture(scope='function')
+def network1(client1, group1, session_env1, session_id):
+    network = create_network(
```
Comment (noah-paige) [inline tests_new/integration_tests/core/vpc/conftest.py:7]: minor: should we wrap this in `try/finally` block similar to some other fixtures to handle the case of if network is not created successfully?

File: tests_new/integration_tests/core/vpc/test_vpc.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+from assertpy import assert_that
+
+from integration_tests.errors import GqlError
+from integration_tests.core.vpc.queries import create_network, delete_network, list_environment_networks
+
+
+def test_create_network(network1, session_id):
+    assert_that(network1).contains_entry(label='testVpc1', tags=[session_id], vpcId='someId')
```
Comment (noah-paige) [inline tests_new/integration_tests/core/vpc/test_vpc.py:8]: needs to `VpcId` (capital V)

File: tests_new/integration_tests/core/vpc/test_vpc.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    assert_that(delete_network).raises(GqlError).when_called_with(
+        client2,
+        vpc_uri=network1.vpcUri,
+    ).contains('UnauthorizedOperation', 'DELETE_NETWORK', network1.vpcUri)
+
+
+def test_list_environment_networks(client1, network1, session_env1, session_id):
+    response = list_environment_networks(client1, environment_uri=session_env1.environmentUri, term=session_id)
+    assert_that(response.count).is_equal_to(1)
+    assert_that(response.nodes[0]).contains_entry(label='testVpc1', vpcId='someId', vpcUri=network1.vpcUri)
```
Comment (noah-paige) [inline tests_new/integration_tests/core/vpc/test_vpc.py:61]: for `listEnvironmentNetworks` had to make the following changes to get working:
- `vpcId` to `VpcId`
- `listEnvironmentNetworks` in `tests_new/integration_tests/core/vpc/queries.py` return statement to `return response.data.listEnvironmentNetworks` 
- `query_environment_networks` at `dataall/backend/dataall/core/vpc/db/vpc_repositories.py` to also filter by tags similar to filter term

**Review by noah-paige — COMMENTED** [review]
left some comments on changes required for tests to work on my end

File: tests_new/integration_tests/core/vpc/test_vpc.py (hunk 3 above)
Comment (dlpzx) [reply tests_new/integration_tests/core/vpc/test_vpc.py:61]: Forgot to push the latest changes! Sorry for this!

Comment (dlpzx) [reply tests_new/integration_tests/core/vpc/test_vpc.py:61]: `query_environment_networks` is a great catch

**Review by noah-paige — APPROVED** [review]
tested and looks good - approving!

**Pull request merged**